packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -1910,3 +1923,437 @@ async fn nested_enums_are_correctly_encoded_decoded() {
 
     assert_eq!(response.value, expected_none);
 }
+
+#[tokio::test]
+async fn can_call_no_arg_predicate_returns_true() {
```

Comment:
Here I feel like you can definitely "group" tests together, because there seems to be a lot of boilerplate code being repeated.

Reply:
So our current workflow is that `unit tests` are put in the file that they are testing. And the harness is for "end to end" tests. 
For reuse of code you can look at other examples in the `harness.rs` file, basically you can skip the whole node/coins/wallet setup and do it just once at the beginning.